src/Editor.jsx

```diff
@@ -36,6 +36,7 @@ class Editor extends Component {
                     inGraph: false,
                 });
             } else {
+                this.setState(defaultGraph);
```

Comment:
So in this case, we want to merge the `defaultGraph` object with a few custom values. I would do something like this:
```
// Copy defaultGraph object
let newState = Object.assign({}, defaultGraph);
newState.step = 1;
newState.gType = (this.state.gType !== null ? this.state.gType : this.state.lastGraphVisited);
newState.inGraph = true;

this.setState(newState);
```

Reply:
I can rewrite it, but Isn't it effectively the same?

Reply:
Strangely enough, you version doesn't work. Not sure why

Reply:
Never mind your version is working. Not sure why it failed, maybe a typo.

Reply:
My version is different because there is only a single setState call - with the previous version, `defaultGraph`'s values may actually be set after the custom values being declared on the next line, overwriting them.